File: legacy-python/envx/config/paths.py
"""Path helpers for project-local envx data."""

from __future__ import annotations

from dataclasses import dataclass
from pathlib import Path


LEGACY_KEY_FILENAME = "key.key"


@dataclass(frozen=True, slots=True)
class ProjectPaths:
    """Resolved filesystem paths for envx storage."""

    root: Path
    config_dir: Path
    config_path: Path
    backup_path: Path
    lock_path: Path
    key_path: Path
    legacy_key_path: Path


def resolve_project_paths(base_dir: Path | None = None) -> ProjectPaths:
    """Resolve the current project's envx paths."""
    root = (base_dir or Path.cwd()).resolve()
    config_dir = root / ".envx"
    return ProjectPaths(
        root=root,
        config_dir=config_dir,
        config_path=config_dir / "config.json",
        backup_path=config_dir / "config.backup.json",
        lock_path=config_dir / "config.lock",
        key_path=config_dir / "key.bin",
        legacy_key_path=config_dir / LEGACY_KEY_FILENAME,
    )

File: internal/config/paths.go
package config

import (
	"path/filepath"
)

const LegacyKeyFilename = "key.key"

type ProjectPaths struct {
	Root          string
	ConfigDir     string
	ConfigPath    string
	BackupPath    string
	LockPath      string
	KeyPath       string
	LegacyKeyPath string
}

func ResolveProjectPaths(baseDir string) ProjectPaths {
	root, _ := filepath.Abs(baseDir)
	configDir := filepath.Join(root, ".envx")
	return ProjectPaths{
		Root:          root,
		ConfigDir:     configDir,
		ConfigPath:    filepath.Join(configDir, "config.json"),
		BackupPath:    filepath.Join(configDir, "config.backup.json"),
		LockPath:      filepath.Join(configDir, "config.lock"),
		KeyPath:       filepath.Join(configDir, "key.bin"),
		LegacyKeyPath: filepath.Join(configDir, LegacyKeyFilename),
	}
}
